app/controllers/challenge_partnerships_controller.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+class ChallengePartnershipsController < ApplicationController
+  before_action :set_form, only: :create
+
+  def show
+    set_form
+  rescue TokenExpiredError
+    redirect_to link_expired_challenge_partnership_path
+  rescue AlreadyChallengedError
+    redirect_to already_challenged_challenge_partnership_path(school_name: @school_name)
+  end
```

Comment:
should we worry about these exceptions in `create` too?

Reply:
I don't think so. The odds of something changing between the two is so slim that I think a 500 is fine here